Revert automatic bpfman image loading into KIND cluster

frobware opened the pull request:
Previously, the Makefile was modified to load the bpfman image
unconditionally into the KIND cluster. This commit reverts that
change. To load the bpfman image into the KIND cluster, use the
`./hack/kind-load-image.sh` script as a discrete step.

For example:
```sh

$ export BPFMAN_IMG=docker.io/library/bpfman:latest

$ cd bpfman
$ docker build -f Containerfile.bpfman.local -t $BPFMAN_IMG . 

$ cd bpfman-operator
$ make run-on-kind

# Note the current image.
$ echo "$(kubectl get -n bpfman daemonsets.apps bpfman-daemon -o jsonpath='{.spec.template.spec.containers[?(@.name=="bpfman")].image}')"
quay.io/bpfman/bpfman:latest

$ ./hack/kind-load-image.sh -v bpfman-deployment $BPFMAN_IMG
docker save  --output "/tmp/bpfman-5G4T1G/image-1.tar" "docker.io/library/bpfman:latest"
kind load image-archive --name "bpfman-deployment" "/tmp/bpfman-5G4T1G/image-1.tar"

$ kubectl set image -n bpfman daemonset/bpfman-daemon bpfman=$BPFMAN_IMG
daemonset.apps/bpfman-daemon image updated

$ kubectl rollout status -n bpfman daemonset/bpfman-daemon
daemon set "bpfman-daemon" successfully rolled out

# Verify that we are running the new image.
$ echo "$(kubectl get -n bpfman daemonsets.apps bpfman-daemon -o jsonpath='{.spec.template.spec.containers[?(@.name=="bpfman")].image}')"
docker.io/library/bpfman:latest

# Notice the two bpfman images we now have in the KIND cluster
$ docker exec -it bpfman-deployment-control-plane /bin/bash
root@bpfman-deployment-control-plane:/# crictl images | grep bpfman
docker.io/library/bpfman-agent                  latest               b61f99a39fca8       501MB
docker.io/library/bpfman-operator               latest               521c4ad502091       180MB
docker.io/library/bpfman                        latest               b16a909169931       405MB
quay.io/bpfman/bpfman                           latest               b13146c64b45f       56.2MB
quay.io/bpfman/csi-node-driver-registrar        v2.9.0               50013f94a28d1       10.6MB
```

This reverts:
- https://github.com/bpfman/bpfman-operator/pull/106
- https://github.com/bpfman/bpfman-operator/pull/111

anfredette reviewed (commented):
With these changes reverted, does `make run-on-kind` build and load the local `bpfman-operator` and `bpfman-agent` images by default?

I was testing w/Mohammed's pr #102 yesterday, and that didn't seem to be happening.  I had to explicitly load upstream images using:
```
BPFMAN_IMG="quay.io/afredette/bpfman:latest" BPFMAN_OPERATOR_IMG="quay.io/mmahmoud/bpfman-opertor:tcx-support" BPFMAN_AGENT_IMG="quay.io/mmahmoud/bpfman-agent:tcx-support" make run-on-kind
```

anfredette commented:
Also, can you spell out the steps needed for someone to use a locally built `bpfman` image as opposed to a remote image with the same name and tag?

anfredette commented:
I was thinking of something like the following:

By default, `make run-on-kind` 
- builds and uses local images for `bpfman-operator` and `bpfman-agent`
- uses the remote version of `quay.io/bpfman/bpfman:latest`

`BPFMAN_LOCAL=1 make run-on-kind`
- uses the local version of whatever `BPFMAN_IMG` is set to
- by default, this is the local version of `quay.io/bpfman/bpfman:latest`

`REMOTE=1 make run-on-kind`
- use the remote versions of everything
- This allows someone to quickly spin up a cluster w/bpfman without building anything for kicking the tires or testing.

anfredette reviewed (changes requested):
Looks good other than the one comment.

anfredette commented:
> With these changes reverted, does `make run-on-kind` build and load the local `bpfman-operator` and `bpfman-agent` images by default?

FYI, I tested this pr and it does, though the suggested change is needed to load the `bpfman-agent` image.

frobware commented on Makefile:
```diff
@@ -354,29 +354,8 @@ push-images: ## Push bpfman-agent and bpfman-operator images.
-	  fi; \
-	done
+load-images-kind: ## Load bpfman-agent, and bpfman-operator images into the running local kind devel cluster.
+	./hack/kind-load-image.sh ${KIND_CLUSTER_NAME} ${BPFMAN_OPERATOR_IMG}
```
I dropped the commit.

frobware commented:
> UPDATE: I added issue #118 for these suggestions to be considered separately.

Noted. I dropped my [testing] commit. 

> Also, can you spell out the steps needed for someone to use a locally built `bpfman` image as opposed to a remote image with the same name and tag?

I updated the PR description, though perhaps the cited example should go in our docs.

anfredette commented:
> I updated the PR description, though perhaps the cited example should go in our docs.

I'm going to approve this pr as is, but please add that info to the docs.  Somewhere in the following area is probably a good place:

https://github.com/bpfman/bpfman/blob/9ae469f2073a45ba342af8a28220630b3672932d/docs/developer-guide/develop-operator.md#L252

anfredette reviewed (approved):
LGTM
